src/support/ios.c

```diff
@@ -648,9 +648,9 @@ int ios_flush(ios_t *s)
     return 0;
 }
 
-void ios_close(ios_t *s)
+int ios_close(ios_t *s)
 {
-    ios_flush(s);
+    int err = ios_flush(s);
     if (s->fd != -1 && s->ownfd)
         close(s->fd);
```

Comment:
check the return value of `close` also. Some file systems don't sync the buffers to disk until the fd is closed even with a preceding flush.